### ISSUE TITLE
Allowing microphone permission in runtime has no effect the first time

### ISSUE THREAD

**Issue by vjlomocso** [xamarin/xamarin-android#2783]
### Steps to Reproduce

1. Request permission using API
2. Allow permission
3. call `AudioRecord#StartRecording`
4. The AudioRecord#StartRecording method will throw an exception saying something like the AudioRecord is uninitialized. This is exactly what happens when you don't allow the permission.

Here's a repro project
https://github.com/vjlomocso/XamarinAndroidPermissionBug

### Expected Behavior
Should not throw exception after allowing permission

### Actual Behavior
Throws exception

### Version Information

https://github.com/vjlomocso/XamarinAndroidPermissionBug/blob/master/versioninfo.txt

### Log File

https://github.com/vjlomocso/XamarinAndroidPermissionBug/blob/master/logcat.log

**Comment by dellis1972** [comment]
I believe this is expected behaviour. On Android 6.0+ the permissions in the manifest are NOT used, they are for older versions of android. The new system you need to request the permission from the user while the app is running. Take a look at the following blog post for details https://blog.xamarin.com/requesting-runtime-permissions-in-android-marshmallow/

I'll close this for now as it is expected. However if you still have a problem after implementing the advice from the link above please reopen.

**Comment by vjlomocso** [comment]
Hi @dellis1972,
Thanks for the response. But I don't think you understand the issue. I'm not even concerned about whether it was in the manifest or not, and I know I need to ask the user for permission at runtime.

The problem is something that doesn't happen in native (Java) Android code, even if it's the exact java equivalent of my repro project. The problem is that even if the user allows the permission at runtime, the microphone still can't be used. I need to close the app and start it again, before I can start using the microphone.

**Comment by dellis1972** [comment]
I repo'd the issue. Will investigate.

**Comment by dellis1972** [comment]
Found the problem. 

```
_recorder = new AudioRecord (AudioSource.Mic, _sampleRate, Channel, AudioEncoding, _bufferSize);
```

this is being called in `Init`. The problem is that is being called BEFORE the permission is granted, so its not being done correctly. This is why it works on app restart.

Moving that code into the `Start` method (and removing the readonly on the property). Fixes the issue. 
I also added a dispose in the `Stop` method to clean up.  see https://gist.github.com/dellis1972/1e5279237f6b65045a74b3c50074e4a6
